build-system/tasks/json-check.js

```diff
@@ -21,6 +21,33 @@ const log = require('fancy-log');
 const through2 = require('through2');
 const {jsonGlobs} = require('../config');
 
+const expectedCaches = ['cloudflare', 'google'];
+
+/**
+ * Fail if caches.json is missing some expected caches.
+ */
+function checkCachesJson() {
+  return gulp.src(['caches.json',])
+      .pipe(through2.obj(function(file) {
+        try {
+          const obj = JSON.parse(file.contents.toString());
+        } catch (e) {
+          // Do nothing since this is handled in checkValidJson.
+        }
+        let foundCaches = [];
+        for (var i = 0; i < obj.caches.length; i++) {
+          foundCaches.push(obj.caches[i].id);
+        }
+        for (const cache of expectedCaches) {
+          if (!cache in foundCaches) {
```

Comment:
This doesn't do what you think it does - it checks whether `!cache` ("the boolean negative of the value in the variable cache") is an _existing index_ of the array foundCaches.

You probably want this to say: `if (!foundCaches.includes(cache))`